### ISSUE TITLE
[VSCode extension] Don't create "lock files" in the user's project folders

### ISSUE THREAD

**Issue by anthonyvdotbe** [apache/netbeans#4962]
### Apache NetBeans version

Apache NetBeans 15

### What happened

NetBeans creates files named `.LCK<filename>~` in my project folders. I don't know the purpose of those lock files, but I doubt VS Code or other tools will recognize/respect those lock files. And if this is solely for NetBeans' internals, I'd expect it to create those lock files in its own private extension storage somewhere, not in my project folders.

### How to reproduce

Opening a `pom.xml` file should be sufficient to get an accompanying `.LCKpom.xml~` to appear.

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Windows 11

### JDK

openjdk 20-ea+22-1594 from https://jdk.java.net/20/

### Apache NetBeans packaging

Apache VSNetBeans for VSCode

### Anything else

FWIW, a quick search reveals that the code related to lock files is in the package [`org.netbeans.modules.masterfs.filebasedfs.fileobjects`](https://github.com/apache/netbeans/blob/master/platform/masterfs/src/org/netbeans/modules/masterfs/filebasedfs/fileobjects/).

### Are you willing to submit a pull request?

Yes

### Code of Conduct

Yes

**Comment by MartinBalin** [comment]
Sorry, I cannot reproduce with VSNetBeans 15.0.301. Opened Maven based project in VSCode, then opened pom.xml, even tried to edit and I cannot see .LCK... file.

**Comment by anthonyvdotbe** [comment]
Strange, I'm using the same version and it happens for me all the time. There's nothing of interest in the output of the language server though. AFAICT there's no setting to enable debug output?

**Comment by ebarboni** [comment]
are you privilegied user on windows ? do you use special folder ? I cannot reproduce too. :/

**Comment by anthonyvdotbe** [comment]
> are you privilegied user on windows ? do you use special folder ? I cannot reproduce too. :/

Yes, I have admin rights. No, the Maven project is in an ordinary folder and there's nothing special about my VS Code installation or workspace.

What would be helpful imho, would be to have a custom build which adds something like:

```java
if(Objects.equals(file.getName(), "pom.xml")) {
    Thread.dumpStack();
}
```

to [LockForFile::getLockFile](https://github.com/apache/netbeans/blob/master/platform/masterfs/src/org/netbeans/modules/masterfs/filebasedfs/fileobjects/LockForFile.java#L236)

and similar for [WriteLockUtils::writeContentOfLckFile](https://github.com/apache/netbeans/blob/master/platform/masterfs/src/org/netbeans/modules/masterfs/filebasedfs/fileobjects/WriteLockUtils.java#L85).

That way I could provide a stacktrace, which would likely be sufficient to understand what's going on without being able to reproduce yourself.

**Comment by lahodaj** [comment]
So, I was able to reproduce the creation of the lock files. I used two distinct `pom.xml` files for that, opened both, and modified both, and then they get the lock files (both of them, for me).

This was on Linux. I couldn't try on Windows yet, but there's a chance this would happen even without the manual modification on Windows, as the files may be internally marked as modified due to different line endings, or alike. Not sure about that.

My understanding  of the cause so far is that files are locked when they are modified, but the lock is not materialized as a file in most cases. But when two files with the same name (not path) are modified, the locks are materialized as files. I was not able to find out what is the reason for this behavior.
